dask/dataframe/multi.py

```diff
@@ -440,7 +440,7 @@ def single_partition_join(left, right, **kwargs):
         align_dataframes=False,
         **kwargs,
     )
-    joined.divisions = divisions
+    joined.divisions = tuple(divisions)
```

Comment:
Interesting. The divisions objects here weren't actually changed by https://github.com/dask/dask/pull/8341 at all. The issue is just that in the code before, we called `new_dd_object`, which called into the `_Frame` constructor, which _does_ coerce `divisions` passed in to a tuple: https://github.com/dask/dask/blob/6c5b4828444393f1974323b0462b8a19fab90c39/dask/dataframe/core.py#L325

Another vote in favor of a setter to continue this pattern!